packages/@aws-cdk/aws-cloud9/lib/environment.ts

```diff
@@ -61,6 +62,29 @@ export interface Ec2EnvironmentProps {
    * @default - no description
    */
   readonly description?: string;
+
+  /**
+   * The AWS CodeCommit repository to be cloned
+   *
+   * @default - do not clone any repository
+   */
+  readonly repositories?: IRepository[];
+}
+
+/**
+ * The interface of the codecommit repository for Cloud9 environment
+ */
+export interface IRepository {
+  /**
+   * AWS CodeCommit repository
+   */
+  readonly repository: codecommit.IRepository,
```

Comment:
2 things:

1. I would probably call this `codeCommitRepository` (given the type).
2. Is there any chance Cloud9 will support other types of repositories in the future (GitHub comes to mind)? Because if the answer is 'yes', the current design will not support that. There are ways to 'future proof' it a little bit so that it will support more types if necessary - just gauging the need for that here :).

Reply:
Thanks for bringing it up. It's very likely to support other types of repos in the future. Let me propose a new design and please review again then.

Reply:
Trying to propose this design. By the time Cloud9 supports other providers such as Github, we should be able to render it accordingly. What do you think?

```ts
/**
 * Repository provider
 */
export enum Cloud9RepositoryProvider {
  /**
   * AWS CodeCommit
   */
  AWS_CODECOMMIT
}

/**
 * The interface of the codecommit repository for Cloud9 environment
 */
export interface Cloud9Repository {
  /**
   * Repository provider
   * @default AWS_CODECOMMIT
   */
  readonly provider?: Cloud9RepositoryProvider;

  /**
   * Repository URL
   */
  readonly url: string;

  /**
   * The path within the development environment's default file system location to clone the AWS CodeCommit
   * repository into. For example, `/REPOSITORY_NAME` would clone the repository into the
   * `/home/USER_NAME/environment/REPOSITORY_NAME` directory in the environment.
   */
  readonly clonePath: string;
}
```